grudge/discretization.py

```diff
@@ -636,17 +563,166 @@ def normal(self, dd):
 
     # }}}
 
+# }}}
+
+
+# {{{ Discretization construction routines
+
+def make_discretization_collection(
+        array_context: ArrayContext,
+        discr_context,
+        order=None,
+        discr_tag_to_group_factory=None,
+        mpi_communicator=None) -> DiscretizationCollection:
+    """Construct a discretization collection given an array context *array_context*
+    and user-provided discretization parameters.
+
+    :arg discr_context: A :class:`~meshmode.mesh.Mesh` object to define
+        the discretization collection over.
+    :arg discr_tag_to_group_factory: A mapping from discretization tags
+        (typically one of: :class:`grudge.dof_desc.DISCR_TAG_BASE`,
+        :class:`grudge.dof_desc.DISCR_TAG_MODAL`, or
```

Comment:
Is overriding the meaning of "modal" allowed?

Reply:
I think this is addressed in https://github.com/inducer/grudge/commit/fa9a8b904134bd46c7ed6ceef4b656d98c76d467. A default modal factory is provided whenever it is not explicitly provided by the user (in a similar way to how there is a default simplex group factor)